### PR TITLE
[Impeller] add triangle fan support and remove drawVertices copying.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Vulkan and GLES backends support triangle fans, so only "un-fan" these geometries when necessary. Now that dispatch gives the vertices geometry as a shared_ptr, we can hold onto that directly without performing another copy.

### CODE REVIEW

File: impeller/renderer/backend/metal/formats_mtl.h [hunk 1]
```diff
@@ -159,6 +159,9 @@ constexpr MTLPrimitiveType ToMTLPrimitiveType(PrimitiveType type) {
       return MTLPrimitiveTypeLineStrip;
     case PrimitiveType::kPoint:
       return MTLPrimitiveTypePoint;
+    case PrimitiveType::kTriangleFan:
+      // UNSUPPORTED
```
Comment (chinmaygarde) [inline impeller/renderer/backend/metal/formats_mtl.h:163]: Can we update this comment to say that callers are expected to perform a cap check.

File: impeller/display_list/dl_vertices_geometry.h [hunk 2]
```diff
@@ -6,13 +6,54 @@
...
+  /// @brief Check if index normalization is required, returning whether or
+  ///        not it was performed.
+  ///
+  /// If true, [indices_] should be used in place of the vertices object's
+  /// indices.
+  bool MaybePerformIndexNormalization(const ContentContext& renderer);
+
+  const std::shared_ptr<const flutter::DlVertices> vertices_;
+  std::vector<uint16_t> indices_;
+  bool performed_normalization_;
```
Comment (chinmaygarde) [inline impeller/display_list/dl_vertices_geometry.h:54]: Zero oout the scalar.

Comment (jonahwilliams) [reply impeller/display_list/dl_vertices_geometry.h]: Done

File: impeller/renderer/backend/metal/formats_mtl.h (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/backend/metal/formats_mtl.h]: Done

**Pull request merged**